### ISSUE TITLE
can't send and receive for the same time

### ISSUE THREAD

**Issue by sunnyreallyCool** [rabbitmq/rabbitmq-server#1452]
## What resource is limited when receiving and sending the message in one thread at mean time?

*Situation:* I was using the rabbitmq in both pull and push mode. First a producer would send a message to push_Q . Then a consumer would listen to the push_Q and have the message immediately. In the same thread, the consumer need to put the message into pull_Q and then convert and get the message from the pull_Q. However. the consumer couldn't get the message and the pull_Q has no message only after it finished the thread. Before it finishes the thread, push_Q always has 1 unacks, and the pull_Q has 0 ready messages. After it finishes the thread, the push_Q has 0 unacks. and the pull_Q has 1 ready message.

## Problem

I was using springamqp in 1.3.6 version. 

* when in the consumer thread, i already traced the code to send the message. and the basic publish method has been invoked successfully. however there is no message in pull_Q, why?
* when i set different connection factories for the different queues, there will be no problems, why?

## sample code
	public void handler(Object obj) {
		Sche1000Message message = (Sche1000Message) obj;
		if (running.compareAndSet(false, true)) {
			if (null == timeFlagHashMap.put(message.getTaskCode(), true)) { 
				pullMessageProducer.sendScheduler((Sche1000Message)message); 
			}
		} else if (timeFlagHashMap.contains(message.getTaskCode())) {
			return;
		} else {
			if (null == timeFlagHashMap.putIfAbsent(message.getTaskCode(), true)) {
				synchronized (this) {
					pullMessageProducer.sendScheduler((Sche1000Message)message);
				}
				if (!running.compareAndSet(false, true)) {
					return;
				}

			} else {
				return;
			}
		}
		dealCachedTask();
	}

	public void dealCachedTask() {
		while (true) {
			Object message;
			synchronized (this) { 
				message = consumer.getMessage();
			}
			timeFlagHashMap.remove(((Sche1000Message) message).getTaskCode());
		}
	}

**Comment by michaelklishin** [comment]
Thank you for your time.

Team RabbitMQ uses GitHub issues for specific actionable items engineers can work on. This assumes two things:

1. GitHub issues are not used for questions, investigations, root cause analysis, discussions of potential issues, etc (as defined by this team)
2. We have a certain amount of information to work with

We get at least a dozen of questions through various venues every single day, often quite light on details.
At that rate GitHub issues can very quickly turn into a something impossible to navigate and make sense of even for our team. Because of that questions, investigations, root cause analysis, discussions of potential features are all considered to be mailing list material by our team. Please post this to [rabbitmq-users](https://groups.google.com/forum/#!forum/rabbitmq-users).

Getting all the details necessary to reproduce an issue, make a conclusion or even form a hypothesis about what's happening can take a fair amount of time. Our team is multiple orders of magnitude smaller than the RabbitMQ community. Please help others help you by providing a way to reproduce the behavior you're
observing, or at least sharing as much relevant information as possible [on the list](https://groups.google.com/forum/#!forum/rabbitmq-users):

 * Server, client library and plugin (if applicable) versions used
 * Server logs
 * A code example or terminal transcript that can be used to reproduce
 * **Full** exception stack traces (not a single line message)
 * `rabbitmqctl status` (and, if possible, `rabbitmqctl environment` output)
 * Other relevant things about the environment and workload, e.g. [a traffic capture](https://www.rabbitmq.com/amqp-wireshark.html)

Feel free to edit out hostnames and other potentially sensitive information.

When/if we have enough details and evidence we'd be happy to file a new issue.

Thank you.

**Comment by michaelklishin** [comment]
Spring AMQP folks take questions on Stack Overflow labelled as `spring-amqp`. See server logs and consider taking a [traffic capture](https://www.rabbitmq.com/amqp-wireshark.html). Collect data, don't guess.

There is no limitation in RabbitMQ that prevents clients from publishing and consuming on a single connection (or channel). There are scenarios in which [publishers are blocked](http://www.rabbitmq.com/alarms.html).
